loadgen/generic_executor.go

```diff
@@ -68,34 +71,52 @@ func (g *GenericExecutor) newRun(info ScenarioInfo) (*genericRun, error) {
 // iterations is reached.
 func (g *genericRun) Run(ctx context.Context) error {
 	ctx, cancel := context.WithCancel(ctx)
-	if g.config.Duration > 0 {
-		ctx, cancel = context.WithTimeout(ctx, g.config.Duration)
+	if g.config.Timeout > 0 {
+		g.logger.Debugf("Will timeout after %v", g.config.Timeout)
+		ctx, cancel = context.WithTimeout(ctx, g.config.Timeout)
 	}
 	defer cancel()
 
+	durationCtx := ctx
+	if g.config.Duration > 0 {
+		durationCtx, cancel = context.WithTimeout(ctx, g.config.Duration)
```

Comment:
Reassigning cancel here is technically fine because it's not used later besides the next line, but it's not advisable